Added Documentation to specify that DataFrame.last() needs the index to be sorted to deliver the expected results

aflah02 opened the pull request:
Added Documentation mentioning that DataFrame.last() needs the index to be sorted to deliver the expected results

Haven't yet worked on raising an error will work as advised

- [ ] closes #38000
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

aflah02 committed:
Update generic.py

Added Documentation mentioning that DataFrame.last() needs the index to be sorted to deliver the expected results

pep8speaks commented:
Hello @aflah02! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/generic.py`](https://github.com/pandas-dev/pandas/blob/60e563f689aff107e916e2b05e812cfdc1affa47/pandas/core/generic.py):

> [Line 8444:1](https://github.com/pandas-dev/pandas/blob/60e563f689aff107e916e2b05e812cfdc1affa47/pandas/core/generic.py#L8444): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 8445:89](https://github.com/pandas-dev/pandas/blob/60e563f689aff107e916e2b05e812cfdc1affa47/pandas/core/generic.py#L8445): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (121 > 88 characters)

aflah02 committed:
Update generic.py

Fixed PEP8 Issues

aflah02 committed:
Update generic.py

Fixed PEP 8  Issues

arw2019 commented on pandas/core/generic.py:
```diff
@@ -8442,6 +8442,9 @@ def last(self: FrameOrSeries, offset) -> FrameOrSeries:
         When having a DataFrame with dates as index, this function can
         select the last few rows based on a date offset.
 
+        Note that for the last() method to work, the DatetimeIndex
```
Personal opinion but I'd rather rephrase the paragraph above to make this clear:
```
For a DataFrame with a sorted DatetimeIndex, this function selects the last few rows based on a date offset.
```

aflah02 replied:
Yeah this seems to be better, I wasn't thinking of changing the original section initially but if that's okay then I'll change that too thanks :)

aflah02 replied:
@arw2019 I did make the changes but I was wondering as was commented in the issue earlier as well do we need to raise an error message as well incase the index is not sorted?

arw2019 replied:
> @arw2019 I did make the changes but I was wondering as was commented in the issue earlier as well do we need to raise an error message as well incase the index is not sorted?

It uses binary search under the hood (logarithmic time) so, with no other information vertifying sortedness (linear time) would degrade performance. If sortedness information is stored somehow then I agree we should throw but AFAIK it isn't in general for a DatetimeIndex.

aflah02 replied:
ohkk understood thanks 👍

aflah02 committed:
Update generic.py

As per recommendation changed the description for DataFrame.last() making it more concise

aflah02 committed:
Update generic.py

Removed trailing whitespace to fix PEP 8 Issues

aflah02 commented:
@arw2019 So do I need to do anything more or will it be merged now?

MarcoGorelli commented:
Looks good, thanks @aflah02 (and thanks @arw2019 for reviewing!)

aflah02 commented:
Thanks @MarcoGorelli
